### ISSUE TITLE
--locked installation fails because proc-macro2 1.0.58 doesn't compile on nightly rust

### ISSUE THREAD

**Issue by missingdays** [cargo-generate/cargo-generate#1009]
**To Reproduce**
1. `cargo install cargo-generate --locked`
2. Installation fails with
```
error[E0635]: unknown feature `proc_macro_span_shrink`
  --> /home/builduser/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.58/src/lib.rs:92:30
   |
92 |     feature(proc_macro_span, proc_macro_span_shrink)
   |                              ^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by missingdays** [comment]
`proc-macro2` has been updated to `1.0.66` in `Cargo.lock`, which would fix the problem, but the version with new `Cargo.lock` file hasn't been released since

**Comment by sassman** [comment]
should be fixed in v0.18.4
